ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -75,18 +75,20 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
     view.doOnAttach {
-      @Suppress("UNCHECKED_CAST")
-      viewModel = dataCollectionViewModel.getTaskViewModel(position) as T
+      @Suppress("UNCHECKED_CAST") val vm = dataCollectionViewModel.getTaskViewModel(position) as? T
 
-      taskView.bind(this, viewModel)
-      taskView.addTaskView(onCreateTaskBody(layoutInflater))
+      if (vm != null) {
```

Comment:
Let's use short circuit to avoid unnecessary nesting.  I think you can just do:

```kt
 val vm = dataCollectionViewModel.getTaskViewModel(position) as? T ?: return
```

Reply:
I could in theory, but it would need to be a labelled return, which our static analyzer disallows.

Reply:
Can you override `onAttach` instead of `onViewCreated`? Then you also won't need the labelled return and `view.doOnAttach`

Reply:
cool, I'll do that!

Reply:
switching to onAttach breaks a bunch of tests--something about a navcontroller not being attached. I think(?) it's because we call a bunch of functions that assume the view is available at that point